manifests/init.pp

```diff
@@ -95,11 +95,10 @@
     default => '',
   }
 
-  #$allowed_versions = concat(['system', 'pypy'], $valid_versions)
-  #unless $version =~ Enum[allowed_versions] {
-    #fail("version needs to be within${allowed_versions}")
-    #}
-  validate_re($version, concat(['system', 'pypy'], $valid_versions))
+  $allowed_versions = concat(['system', 'pypy'], $valid_versions)
+  unless $version =~ Enum[$allowed_versions] {
```

Comment:
Should this be `unless $version in $allowed_versions` if it's an array? Not sure what the data type of `$valid_versions` is.

Reply:
`$valid_versions`, when taken from `python::params`, is an array. But even if it would be a string, the `concat()` would generate an array for `$allowed_versions`. I guess one could also use the `in` syntax you suggested. Not sure that the performance difference is.